src/containers/course-section/CourseSectionContainer.tsx

```diff
@@ -188,6 +196,56 @@ const CourseSectionContainer: FC<SectionProps> = ({
     }
   }
 
+  const handleEditAttachment = (params?: UpdateAttachmentParams) =>
+    ResourceService.updateAttachment(params)
+
+  const { fetchData: updateData } = useAxios({
+    service: handleEditAttachment,
+    fetchOnMount: false,
+    onResponse: (attachment: Attachment) => {
+      setResources((prev) =>
+        prev.map((resource) => {
+          if (resource._id === attachment._id) {
+            return { ...resource, ...attachment }
+          }
+          return resource
+        })
+      )
+    }
+  })
+
+  const editResource = (resource: CourseResources) => {
+    const resourceType = resource.resourceType
+
+    if (!resourceType) return
+
+    if (resourceType === ResourcesTypes.Attachments) {
+      openModal({
+        component: (
+          <EditAttachmentModal
+            attachment={resource as Attachment}
+            closeModal={closeModal}
+            updateAttachment={updateData}
+          />
+        )
+      })
+    } else {
+      const navigationFiled = resourceNavigationMap[
+        resourceType
+      ] as keyof typeof authRoutes.myResources
+
+      window
+        .open(
+          createUrlPath(
+            authRoutes.myResources[navigationFiled].path,
+            resource._id
+          ),
+          '_blank'
+        )
+        ?.focus()
```

Comment:
What is `focus` used for here?

Reply:
In order to focus user on new tab (edit quiz or lessons)